torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2046,6 +2086,20 @@ void ProcessGroupNCCL::watchdogHandler() {
       lastStatusUpdateTime = std::chrono::steady_clock::now();
     }
 
+    // Check and set remote error if it has not been set before
+    if (getError() == ErrorType::NO_ERROR) {
+      int remoteErrorRank =
+          getSignalSrcRank(store_, std::string(REMOTE_ERROR_SIGNAL));
```

Comment:
I haven't checked whether `getSignalSrcRank` could be blocking or not. Let's be careful when putting potentially blocking call in watchdog.

Reply:
It's non blocking in a sense, we first 'check' if key exists which is nonblocking, and then read/get the value only if the key exists. Otherwise, if we try to get the kv directly, it could be blocking.